src/main/java/com/launchableinc/ingest/commits/CommitGraphCollector.java

```diff
@@ -407,6 +400,39 @@ That is, find submodules that are available in the working tree (thus `!isBare()
       }
     }
 
+    private void collectFiles(TreeWalk treeWalk, Consumer<VirtualFile> receiver) throws IOException {
+      if (!collectFiles)  return;
+
+      int c = treeWalk.getTreeCount();
+
+      OUTER:
+      while (treeWalk.next()) {
+        ObjectId head = treeWalk.getObjectId(0);
+        for (int i=1; i<c; i++) {
+          if (head.equals(treeWalk.getObjectId(i))) {
+            // file at the head is identical to one of the uninteresting commits,
+            // meaning we have already seen this file/directory on the server.
+            // if it is a dir, there's no need to visit this whole subtree, so skip over
+            continue OUTER;
+          }
+        }
+
+        if (treeWalk.isSubtree()) {
+          treeWalk.enterSubtree();
+        } else {
+          if ((treeWalk.getFileMode(0).getBits()&FileMode.TYPE_MASK)==FileMode.TYPE_FILE) {
```

Comment:
Even if a workspace allows collecting file contents, should we avoid collecting hidden files? WDYT??

Reply:
What is a "hidden file"? `TreeWalk` is only going through files managed in Git. Git only has two types of files -- regular files (permission 644) and executable files (permission 755).

Reply:
There are also hidden files under Git management, right?  
These are part of the results from `git ls-files` in this repo:

```
$ git ls-files
.bazelrc
.bazelversion
.editorconfig
.github/release.yml
.github/workflows/bazel-test.yml
.github/workflows/e2e.yml
...
```

Should we include these as well?  
Is it okay to collect them since they're under Git management..?

Reply:
We briefly discussed about this in a call and decided that this is not an issue atm.